pulsar-jms/src/main/java/com/datastax/oss/pulsar/jms/PulsarConnectionFactory.java

```diff
@@ -1370,6 +1371,26 @@ public int compare(Message o1, Message o2) {
       ((BlockingQueue<Message>) oldQueue).drainTo(newQueue);
 
       incomingMessages.set(c, newQueue);
+
+
+      Field consumersField = MultiTopicsConsumerImpl.class
+              .getDeclaredField("consumers");
+
+      consumersField.setAccessible(true);
+
+      ConcurrentHashMap<String, ConsumerImpl<?>> consumers = (ConcurrentHashMap) consumersField
+          .get(consumerBase);
+         Method setCurrentReceiverQueueSizeMethod = ConsumerImpl.class.getDeclaredMethod("setCurrentReceiverQueueSize", int.class);
+      setCurrentReceiverQueueSizeMethod.setAccessible(true);
+        for (ConsumerImpl<?> consumer : consumers.values()) {
+            String topic = consumer.getTopic();
+            int last = topic.lastIndexOf('-');
+            int prio = Integer.parseInt(topic.substring(last + 1));
+            int size = (prio + 1) * 10;
```

Comment:
TODO: consider the configured receiveQueueSize

this should be something like

TOTAL = 1 + 2 + 3 + 4 ..... + 10
SIZE = max(1, receiveQueueSize * (prio + 1) / TOTAL)